src/System.IO.FileSystem/tests/Performance/Perf.Directory.cs

```diff
@@ -58,33 +58,54 @@ public void Exists()
             Directory.Delete(testFile);
         }
 
-        [Benchmark]
-        [InlineData(10, 100)]
-        [InlineData(100, 10)]
-        [InlineData(1000, 1)]
-        [OuterLoop("Takes a lot of time to finish")]
-        public void RecursiveCreateDirectory(int depth, int times)
+        public string GetTestDeepFilePath(int depth)
         {
-            if (!PlatformDetection.IsWindows && depth == 1000)
-                return;
-            
-            // Setup
-            string rootDirectory = GetTestFilePath();
-            StringBuilder sb = new StringBuilder(5000);
+            string directory = Path.DirectorySeparatorChar + "a";
+            StringBuilder sb = new StringBuilder(depth * 2);
             for (int i = 0; i < depth; i++)
             {
-                sb.Append(@"\a");
+                sb.Append(directory);
             }
 
-            string path = sb.ToString();
+            return sb.ToString();
+        }
+
+        public static TheoryData<int, int> RecursiveDepthData
+        {
+            get
+            {
+                var data = new TheoryData<int, int>();
+                data.Add(10, 100);
+
+                // Length of the path can be Atmost 260 characters on netfx.
+                if (PlatformDetection.IsNetCore)
+                {
+                    data.Add(100, 10);
+                    // Length of the path can be Atmost 4096 characters on Unix. 
```

Comment:
It isn't 4096 characters- it is either 1024 or 4096 UTF-8 bytes. Mac is 4096 afaik and _should_ do this ok.

Reply:
I tried it on mac. it gives an error for path being to long on mac for 1000

Reply:
u mean ```1024 characters or 4096 UTF-8 Bytes```

Reply:
```
                // Length of the path can be 260 characters on netfx.
                if (AreAllLongPathsAvailable)
                {
                    data.Add(100, 10);
                    // Length of the path can be at most 4096 UTF-8 bytes on Unix. 
                    if (PlatformDetection.IsWindows)
                        data.Add(1000, 1);
                }

```
is this okay ?

Reply:
// Most Unix distributions have a maximum path length of 1024 characters (1024 UTF-8 bytes)